src/app/views/staff/staff.controller.coffee

```diff
@@ -22,13 +24,27 @@
 
   vm.updateStaff = ->
     vm.isSaving = true
-    MnoeUsers.updateStaff(vm.staff).then(
-      () ->
-        toastr.success("mnoe_admin_panel.dashboard.staff.update_staff.toastr_success", {extraData: { staff_name: "#{vm.staff.name} #{vm.staff.surname}"}})
-      (error) ->
-        toastr.error("mnoe_admin_panel.dashboard.staff.update_staff.toastr_error", {extraData: { staff_name: "#{vm.staff.name} #{vm.staff.surname}"}})
-        $log.error("An error occurred while updating staff:", error)
-    ).finally(-> vm.isSaving = false)
+
+    updateStaffAction = ->
+      MnoeUsers.updateStaff(vm.staff).then(
+        (response) ->
+          vm.staff = response.data.user
+          vm.staff.admin_role_was = vm.staff.admin_role
+          toastr.success("mnoe_admin_panel.dashboard.staff.update_staff.toastr_success", {extraData: { staff_name: "#{vm.staff.name} #{vm.staff.surname}"}})
+        (error) ->
+          toastr.error("mnoe_admin_panel.dashboard.staff.update_staff.toastr_error", {extraData: { staff_name: "#{vm.staff.name} #{vm.staff.surname}"}})
+          $log.error("An error occurred while updating staff:", error)
+      ).finally(-> vm.isSaving = false)
+
+    if vm.staff.admin_role_was == 'staff' &&  vm.staff.admin_role != 'staff' && vm.staff.client_ids.length
+      modalOptions =
+        closeButtonText: 'mnoe_admin_panel.dashboard.staff.update_staff_role.cancel'
+        actionButtonText: 'mnoe_admin_panel.dashboard.staff.update_staff_role.action'
+        headerText: 'mnoe_admin_panel.dashboard.staff.update_staff_role.proceed'
+        bodyText: 'mnoe_admin_panel.dashboard.staff.update_staff_role.perform'
+      MnoConfirm.showModal(modalOptions).then(updateStaffAction).finally(-> vm.isSaving = false)
+    else
+      updateStaffAction()
```

Comment:
`updateStaffAction()` is only used here, maybe you should implement it directly here.